renderer/components/editor/video/index.js

```diff
@@ -0,0 +1,313 @@
+};
+
+const PlayBar = ({skip, startTime, duration = 0, previewDuration, currentTime = 0, scale = 1}) => {
+  const left = startTime * scale;
+  const roundedLeft = startTime === 0 ? '' : 'play-bar--rounded';
+  return (
+    <React.Fragment>
+      <div className="play-bar play-bar--background" style={{opacity: duration === previewDuration ? 0 : 1}}/>
+      <div className="play-bar play-bar--playable" style={{
+        width: `${(previewDuration * scale)}px`,
+        left: `${left}px`
+      }}/>
+      <div
+        className={`play-bar play-bar--current-time ${roundedLeft}`}
+        style={{
+          width: `${(currentTime * scale)}px`,
+          left: `${left}px`
+        }}
+      />
+      <div onClick={event => {
+        const time = getTimestampAtEvent(event, duration);
+        skip(time);
+      }} className="play-bar play-bar--clickarea"/>
+      <style jsx>{
+        `    
+        .play-bar {
+          position: absolute;
+          height: 4px;
+          left: 0;
+          bottom: 0;
+          border-radius: 0px 2px 2px 0px;
+          pointer-events: none;
+        }
+        .play-bar--clickarea {
+          pointer-events: auto;
+          -webkit-app-region: no-drag;
+          height: 30px;
+          transform: translateY(50%);
+          z-index: 100;
+          width: 100%;
+        }
+        .play-bar--current-time {
+          background: linear-gradient(90deg, #7146FE 0%, #00FFBE 100%);
+        }
+        .play-bar--background {
+          box-shadow: 0 1px 2px rgba(0,0,0,.1);
+          transition: opacity 200ms ease;
+          background: rgba(255,255,255,.40);
+          width: 100%;
+        }
+        .play-bar--playable {
+          background: rgba(255,255,255,.40);
+          width: 100%;
+        }
+        .play-bar--rounded {
+          border-radius: 2px 2px 2px 2px;
+        }
+        `
+      }</style>
+    </React.Fragment>
+  );
+};
+
+PlayBar.propTypes = {
+  skip: PropTypes.func.isRequired,
+  duration: PropTypes.number,
+  currentTime: PropTypes.number,
+  scale: PropTypes.number,
+  startTime: PropTypes.number,
+  previewDuration: PropTypes.number
+};
+
+export default class Video extends React.Component {
+  static propTypes = {
+    src: PropTypes.string.isRequired
+  }
+
+  state = {currentTime: 0, duration: null, endTime: null, startTime: 0, isPlaying: false, width: 0, hasFocus: false}
+
+  onRef = videoRef => {
+    this.videoRef = videoRef;
+  };
+
+  handleResize = () => this.setState({...this.state, width: window.innerWidth});
+
+  componentDidMount() {
+    this.handleResize();
+    window.addEventListener('resize', this.handleResize);
+  }
+
+  componentWillUnmount() {
+    this.onStop();
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  onEnded = () => {
+    this.videoRef.currentTime = this.state.startTime;
+    this.videoRef.play();
+  }
+
+  onDurationChange = event =>
+    this.setState({
+      duration: event.target.duration,
```

Comment:
We should prefer `currentTarget` over `target` whenever possible, as it's safer.